src/authentication.ts

```diff
@@ -54,14 +55,45 @@ export async function redirectToLogin(
     if (config?.preLogin) config.preLogin()
 
     if (method === 'popup') {
-      const handle: null | WindowProxy = window.open(loginUrl, 'loginPopup', 'popup width=600 height=600')
+      const { left, top, width, height } = calculatePopupPosition(600, 600)
+      const handle: null | WindowProxy = window.open(
+        loginUrl,
+        'loginPopup',
+        `width=${width},height=${height},top=${top},left=${left}`
+      )
       if (handle) return
       console.warn('Popup blocked. Redirecting to login page. Disable popup blocker to use popup login.')
     }
     window.location.assign(loginUrl)
   })
 }
 
+function calculatePopupPosition(popupWidth = 600, popupHeight = 600): TPopupPosition {
+  // Calculate the screen dimensions and position the popup at the center
+  const screenLeft = window.screenLeft || window.screenX
+  const screenTop = window.screenTop || window.screenY
+  const screenWidth = window.innerWidth || document.documentElement.clientWidth || screen.width
+  const screenHeight = window.innerHeight || document.documentElement.clientHeight || screen.height
+
+  // Calculate the position to center the popup
+  let left = screenLeft + (screenWidth - popupWidth) / 2
+  let top = screenTop + (screenHeight - popupHeight) / 2
+
+  // Ensure the bottom-right corner does not go off the screen
+  // Adjust the left and top positions if necessary
+  const maxLeft = screenLeft + (screenWidth - popupWidth)
+  const maxTop = screenTop + (screenHeight - popupHeight)
+
+  left = Math.min(left, maxLeft) // Ensure the right side is within the screen
+  top = Math.min(top, maxTop) // Ensure the bottom side is within the screen
+
+  // Ensure the popup is not positioned off the top/left of the screen
+  left = Math.max(0, left)
+  top = Math.max(0, top)
+
+  return { left, top, width: popupWidth, height: popupHeight }
```

Comment:
This is cool and all, but if this is meant to fix all cases, what about the case where the screen is smaller than the dialog size? Then surely, we should resize the dialog to match the screen?

So

```js
const maxLeft = screenLeft + (screenWidth - popupWidth)
const maxTop = screenTop + (screenHeight - popupHeight)

return {
  left: Math.max(0, Math.min(left, maxLeft)),
  top: Math.max(0, Math.min(top, maxTop)),
  width: Math.min(screenWidth, popupWidth),
  height: Math.min(screenHeight, popupHeight),
}
```

Reply:
Did this myself  ✌️